gax-java/gax/src/main/java/com/google/api/gax/logging/LoggingUtils.java

```diff
@@ -0,0 +1,141 @@
+
+@InternalApi
+public class LoggingUtils {
+
+  private static boolean loggingEnabled = isLoggingEnabled();
+  static final String GOOGLE_SDK_JAVA_LOGGING = "GOOGLE_SDK_JAVA_LOGGING";
+
+  static boolean isLoggingEnabled() {
+    String enableLogging = System.getenv(GOOGLE_SDK_JAVA_LOGGING);
+    return "true".equalsIgnoreCase(enableLogging);
+  }
+
+  /**
+   * Sets logDataBuilder with service name, rpc name, endpoint and request headers based on logging
+   * level
+   *
+   * @param serviceName
+   * @param rpcName
+   * @param endpoint
+   * @param requestHeaders
+   * @param logDataBuilder
+   * @param loggerProvider
+   */
+  public static void recordServiceRpcAndRequestHeaders(
+      String serviceName,
+      String rpcName,
+      String endpoint,
+      Map<String, String> requestHeaders,
+      LogData.Builder logDataBuilder,
+      LoggerProvider loggerProvider) {
+    if (loggingEnabled) {
+      Slf4jLoggingHelpers.recordServiceRpcAndRequestHeaders(
+          serviceName, rpcName, endpoint, requestHeaders, logDataBuilder, loggerProvider);
+    }
+  }
+
+  /**
+   * Sets logDataBuilder with response headers based on logging level
+   *
+   * @param headers
+   * @param logDataBuilder
+   * @param loggerProvider
+   */
+  public static void recordResponseHeaders(
+      Map<String, String> headers, LogData.Builder logDataBuilder, LoggerProvider loggerProvider) {
+    if (loggingEnabled) {
+      Slf4jLoggingHelpers.recordResponseHeaders(headers, logDataBuilder, loggerProvider);
+    }
+  }
+
+  /**
+   * Sets logDataBuilder with respond payload based on logging level
+   *
+   * @param message
+   * @param logDataBuilder
+   * @param loggerProvider
+   * @param <RespT>
+   */
+  public static <RespT> void recordResponsePayload(
+      RespT message, LogData.Builder logDataBuilder, LoggerProvider loggerProvider) {
+    if (loggingEnabled) {
+      Slf4jLoggingHelpers.recordResponsePayload(message, logDataBuilder, loggerProvider);
+    }
+  }
+
+  /**
+   * Log response based on logging level configured
+   *
+   * @param status
+   * @param logDataBuilder
+   * @param loggerProvider
+   */
+  public static void logResponse(
+      String status, LogData.Builder logDataBuilder, LoggerProvider loggerProvider) {
+    if (loggingEnabled) {
+      Slf4jLoggingHelpers.logResponse(status, logDataBuilder, loggerProvider);
+    }
+  }
+
+  /**
+   * Log request based on logging level configured
+   *
+   * @param message
+   * @param logDataBuilder
+   * @param loggerProvider
+   * @param <RespT>
+   */
+  public static <RespT> void logRequest(
+      RespT message, LogData.Builder logDataBuilder, LoggerProvider loggerProvider) {
+    if (loggingEnabled) {
+      Slf4jLoggingHelpers.logRequest(message, logDataBuilder, loggerProvider);
+    }
+  }
+
+  public static void executeWithTryCatch(ThrowingRunnable action) {
+    try {
+      action.run();
+    } catch (Throwable t) {
+      // let logging exceptions fail silently
```

Comment:
I wonder if it would make since to log this error at the lowest levels (maybe TRACE?) about any logging exceptions? 

It would be like using logging to debug logging errors... Which honestly doesn't sound like something people would be doing. Just some thoughts, probably doesn't make any sense.